### PR TITLE
Remove InternalRequest from TransportSingleShardAction

### PR DESCRIPTION
**Pull request by mfussenegger**
Looks like it is not required

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/action/support/single/shard/TransportSingleShardAction.java [hunk 1]
```diff
@@ -156,21 +147,14 @@ private AsyncSingleAction(Request request, ActionListener<Response> listener) {
                 throw blockException;
             }
 
-            String concreteSingleIndex;
-            if (resolveIndex(request)) {
-                concreteSingleIndex = IndexNameExpressionResolver.concreteSingleIndex(clusterState, request).getName();
```
Comment (romseygeek) [inline server/src/main/java/org/elasticsearch/action/support/single/shard/TransportSingleShardAction.java:161]: Has the index name in the request already been resolved earlier?

Comment (mfussenegger) [reply server/src/main/java/org/elasticsearch/action/support/single/shard/TransportSingleShardAction.java:161]: Apparently it's always called with the right index names - not aliases.

This was effectively dead code. All the places where `resolveIndex` was true never accessed the `concreteIndex` property of the `InternalRequest`. There was one place where `concreteIndex()` was used, but that had `resolveIndex()` overriden to return false.

**Review by romseygeek — APPROVED** [review]

**Comment by mfussenegger** [comment]
retest this please

Flaky MetadataTrackerITest

**Comment by mfussenegger** [comment]
retest this please

this time sphinx failed downloading  intersphinx inventory

**Pull request merged**